### PR TITLE
feat(dapps): Implemented hovering and proper image color for dapps item delegate.

### LINKED ISSUE #14760: [UI - dapps Connection] dApps item delegate
[Design](https://www.figma.com/design/HrmZp1y4S77QJezRFRl6ku/dApp-Interactions---Milestone-1?node-id=3617%3A47563&t=onun7PNHBei8WnPS-1)

Screenshot 2024-05-15 at 13 07 35

Screenshot 2024-05-15 at 13 07 43

  - Not highlighted
  - Image + title + subtitle (no badge in image for MVP)
  - `Disconnect` button with tooltip + hover

### PR DESCRIPTION
**Pull request by Seitseman**
Closes #14760

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image
image

### CODE REVIEW

File: ui/imports/shared/popups/walletconnect/DAppsListPopup.qml [hunk 1]
```diff
@@ -87,113 +87,117 @@ Popup {
             model: root.model
             visible: !listPlaceholder.visible
 
-            delegate: DAppDelegate {
+            delegate: MouseArea {
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/DAppsListPopup.qml:90]: Why is the delegate no longer a `DAppDelegate`? I think the goal here should be to adjust that, not replace it with a random `MouseArea`

**Review by caybro — CHANGES_REQUESTED** [review]
The icon of the disconnect button is also visibly not quite centered

File: ui/imports/shared/popups/walletconnect/DAppsListPopup.qml (hunk 1 above)
Comment (alexjba) [reply ui/imports/shared/popups/walletconnect/DAppsListPopup.qml:90]: DAppDelegate was an inline component and it's causing issues on qt 5.15.2. But I agree it should still be a DAppDelegate extracted in a file.

Comment (Seitseman) [reply ui/imports/shared/popups/walletconnect/DAppsListPopup.qml:90]: In the scope of this PR I will move the delegates out to separate files, so there will be no crashes with Qt5.15.2.

File: storybook/qmlTests/tests/tst_DAppsWorkflow.qml [hunk 2]
```diff
@@ -25,630 +25,630 @@ Item {
     width: 600
     height: 400
 
-    // // TODO #15151 fix CI crash and re-enable tests
```
Comment (stefandunca) [inline storybook/qmlTests/tests/tst_DAppsWorkflow.qml:28]: 🙏 awesome fixing the elusive crash on CI!

**Review by stefandunca — APPROVED** [review]
Great seeing the CI crash fixed in tests. Please follow up on @caybro's and this seems good to go.

**Review by alexjba — APPROVED** [review]
Nice! Glad to see the tests up and running! 👍

**Comment by caybro** [comment]
This looks like it needs a rebase :)

**Comment by caybro** [comment]
This still doesn't look right ;)
image

**Comment by alexjba** [comment]
> This still doesn't look right ;) image

It's moving the inline components in different files to unblock the qml tests. I guess it's expected 😄

File: ui/imports/shared/popups/walletconnect/controls/DAppCard.qml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import QtQuick.Layouts 1.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+ColumnLayout {
+        property alias name: appNameText.text
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/controls/DAppCard.qml:12]: Some weird indent here in the whole file

File: ui/imports/shared/popups/walletconnect/panels/ContentPanel.qml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+Rectangle {
+    id: contentPanelRect
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/panels/ContentPanel.qml:8]: We have a convention to use `root` as the toplevel item id

File: ui/imports/shared/popups/walletconnect/panels/EstimatedTimeDisplay.qml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ColumnLayout {
+    id: root
+
+    StatusBaseText {
+        text: qsTr("Est. time:")
+        font.pixelSize: 12
+        color: Theme.palette.directColor1
+    }
+    StatusBaseText {
+        text: root.estimatedTimeText
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/panels/EstimatedTimeDisplay.qml:16]: Where is `estimatedTimeText` coming from?

File: ui/imports/shared/popups/walletconnect/panels/IntentionPanel.qml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                bgBorderWidth: 2
+                bgBorderColor: Theme.palette.statusAppLayout.backgroundColor
+                source: "assets/sign.svg"
+            }
+        }
+    }
+
+    // Names and intentions
+    StatusBaseText {
+        text: qsTr("%1 wants you to %2 with %3").arg(dappName).arg(d.userDisplayNaming).arg(account.name)
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/panels/IntentionPanel.qml:58]: Where is `d.userDisplayNaming` coming from?

File: ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml [hunk 7]
```diff
@@ -1,106 +1,107 @@
...
+        Item {
+            Layout.preferredWidth: 32
+            Layout.preferredHeight: 32
+
+            StatusImage {
+                id: iconImage
+
+                anchors.fill: parent
+
+                source: iconUrl
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml:35]:
```suggestion
                source: root.iconUrl

```

File: ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml [hunk 8]
```diff
@@ -1,106 +1,107 @@
...
+
+            StatusIcon {
+                id: fallbackImage
+
+                anchors.fill: parent
+
+                icon: "dapp"
+                color: Theme.palette.baseColor1
+
+                visible: iconImage.isLoading || iconImage.isError || !iconUrl
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml:47]:
```suggestion
                visible: iconImage.isLoading || iconImage.isError || !root.iconUrl

```

File: ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml [hunk 9]
```diff
@@ -1,106 +1,107 @@
...
+                }
+            }
+        }
+
+        ColumnLayout {
+            Layout.leftMargin: 12
+            Layout.rightMargin: 12
+
+            StatusBaseText {
+                text: name
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml:66]:
```suggestion
                text: root.name

```

File: ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml [hunk 10]
```diff
@@ -1,106 +1,107 @@
...
+
+                font.pixelSize: 13
+                font.bold: true
+
+                elide: Text.ElideRight
+
+                clip: true
+            }
+            StatusBaseText {
+                text: url
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/controls/DAppDelegate.qml:78]:
```suggestion
                text: root.url

```

File: storybook/qmlTests/tests/tst_DAppsWorkflow.qml [hunk 11]
```diff
@@ -25,630 +25,632 @@ Item {
...
-    //     let prevRequests = sdk.getActiveSessionsCallbacks.length
-    //     sdk.sessionRequestEvent(requestEvent)
-    //     // Service might trigger a sessionRequest event following the getActiveSessions call
-    //     let callback = sdk.getActiveSessionsCallbacks[prevRequests].callback
-    //     let session = JSON.parse(Testing.formatApproveSessionResponse([network.chainId, 7], [account.address]))
-    //     callback({"b536a": session})
-
-    //     return {sdk, session, account, network, topic, request}
-    // }
+    // TODO #15151 fix CI crash and re-enable tests
```
Comment (caybro) [inline storybook/qmlTests/tests/tst_DAppsWorkflow.qml:28]: Drop the TODO? :)

File: ui/imports/shared/popups/walletconnect/panels/ContentPanel.qml (hunk 4 above)
Comment (Seitseman) [reply ui/imports/shared/popups/walletconnect/panels/ContentPanel.qml]: Fixed

**Review by caybro — APPROVED** [review]
LGTM

**Pull request merged**